routing_model/src/flows_dst.rs

```diff
@@ -325,20 +388,42 @@ impl CheckAndProcessElderChange {
         state
     }
 
+    fn routine_state(&self) -> &ProcessElderChangeState {
+        match &self
+            .0
+            .check_and_process_elder_change_routine
+            .sub_routine_process_elder_change
+        {
+            Some(state) => state,
+            _ => panic!("Expect ProcessElderChange {:?}", &self),
+        }
+    }
+
+    fn mut_routine_state(&mut self) -> &mut ProcessElderChangeState {
+        let clone = self.clone();
+        match &mut self
+            .0
+            .check_and_process_elder_change_routine
+            .sub_routine_process_elder_change
+        {
+            Some(state) => state,
+            _ => panic!("Expect ProcessElderChange {:?}", &clone),
+        }
+    }
+
+    fn change_elder(&self) -> ChangeElder {
+        self.routine_state().change_elder.clone()
+    }
+
     fn mark_elder_change(&self) -> Self {
         let mut state = self.clone();
 
-        let change_elder = state.mut_routine_state().change_elder.take().unwrap();
+        let change_elder = state.mut_routine_state().change_elder.clone();
```

Comment:
What is it not using change_elder().
Probably better to not have the change_elder() function at all.